stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2729,41 +2901,18 @@ impl NakamotoChainState {
         monitoring::set_last_block_transaction_count(u64::try_from(block.txs.len()).unwrap());
         monitoring::set_last_execution_cost_observed(&block_execution_cost, &block_limit);
 
-        // get previous burn block stats
+        // get previous burn block stats, for the transaction receipt
         let (parent_burn_block_hash, parent_burn_block_height, parent_burn_block_timestamp) =
-            if block.is_first_mined() {
-                (BurnchainHeaderHash([0; 32]), 0, 0)
-            } else {
-                match SortitionDB::get_block_snapshot_consensus(burn_dbconn, &parent_ch)? {
-                    Some(sn) => (
-                        sn.burn_header_hash,
-                        u32::try_from(sn.block_height).map_err(|_| {
-                            ChainstateError::InvalidStacksBlock(
-                                "Burn block height exceeds u32".into(),
-                            )
-                        })?,
-                        sn.burn_header_timestamp,
-                    ),
-                    None => {
-                        // shouldn't happen
-                        warn!(
-                            "CORRUPTION: block {}/{} does not correspond to a burn block",
-                            &parent_ch, &parent_block_hash
-                        );
-                        (BurnchainHeaderHash([0; 32]), 0, 0)
-                    }
-                }
-            };
-
+            Self::get_tenure_burn_block_info(burn_dbconn, block.is_first_mined(), &parent_ch)?;
```

Comment:
The refactored interface of this function is very confusing to me and seems unecessary. `is_first_mined()` is something that only makes sense in the context of `block`, but its passed as a bool to the refactored method, which also returns two unlabeled u64s, and the only thing that this code block is doing is looking up a snapshot and pulling out three fields. Like, we could just make this less verbose:

```rust
            if block.is_first_mined() {
                (BurnchainHeaderHash([0; 32]), 0, 0)
            } else {
                let snapshot = SortitionDB::get_block_snapshot_consensus(burn_dbconn, &parent_ch)?
                    .unwrap_or_else(||
                          warn!("CORRUPTION: {parent_ch} does not correspond to a burn block");
                          ChainstateError::InvalidStacksBlock("No such parent consensus hash")
                    )?;
                (sn.burn_header_hash, sn.block_height, sn.burn_header_timestamp)
            };
```

But I think adding a once-used function (in a separate file) that just shuttles fields out of another function call obfuscates what's happening.

Reply:
The goal here was to cut down on the size of `append_block()`, so different self-contained bits of functionality within it could be unit-tested.

Reply:
I think that this particular case is too aggressive of a separation. All this is doing is invoking a single, already-existent function, and pulling three fields out of the result.

Reply:
I put it back